### PR TITLE
[DUOS-2612] Controlled Access Grants Component

### PR DESCRIPTION
**Pull request by hams7504**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2612

### Summary
- Created a sortable table component
- Used table component to create the controlled access grants section of the user profile

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/SortableTable.js [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+  order: PropTypes.oneOf(['asc', 'desc']).isRequired,
+  orderBy: PropTypes.string.isRequired,
+};
+
+export default function SortableTable(props) {
+
+  const {
+    rows,
+    headCells
+  } = props;
```
Comment (rushtong) [inline src/components/SortableTable.js]: I think this component would be more useful if a predefined `theme` were passed in as a prop from the caller.
As a side note, we do have another `theme.js` class which can seem confusing. I'm wondering if `SortableTable` could be it's own package level (i.e. directory) component with `EnhancedTableHead` and `Themes` as separate components in that package.

File: src/components/SortableTable.js [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+                  const labelId = `enhanced-table-checkbox-${index}`;
+
+                  return (
+                    <TableRow
+                      hover
+                      onClick={(event) => handleClick(event, row.name)}
+                      tabIndex={-1}
+                      key={row.datasetIdentifier}
+                      selected={isItemSelected}
+                    >
```
Comment (rushtong) [inline src/components/SortableTable.js]: minor nit: collapse the closing tag to the previous line. Here and in other places as well.

**Review by rushtong — APPROVED** [review]
Some minor comments inline, but otherwise, this looks great! I can see some potential improvements such as sorting on different columns, but we should only tackle that if it becomes a requirement.

**Comment by hams7504** [comment]
> Some minor comments inline, but otherwise, this looks great! I can see some potential improvements such as sorting on different columns, but we should only tackle that if it becomes a requirement.

Could you explain what you mean by "sorting on different columns"? Currently, the table should be able to sort based on all columns passed in through the headCells variable.

File: src/components/SortableTable.js (hunk 1 above)
Comment (hams7504) [reply src/components/SortableTable.js]: I can implement it that way! I think the SortableTable file is way too crowded.

**Review by fboulnois — COMMENTED** [review]
Once the Codacy issues have been resolved, let's rebase and flatten this PR to make it easier to review.

**Comment by rushtong** [comment]
> > Some minor comments inline, but otherwise, this looks great! I can see some potential improvements such as sorting on different columns, but we should only tackle that if it becomes a requirement.
> 
> Could you explain what you mean by "sorting on different columns"? Currently, the table should be able to sort based on all columns passed in through the headCells variable.

You are correct, I didn't see the additional sorts, sorry about that!

**Pull request closed without merging**